tests/aws/services/stepfunctions/v2/test_stepfunctions_v2.py

```diff
@@ -568,6 +589,52 @@ def _create_sm(*_):
         parallelize(_create_sm, list(range(num_machines)), size=2)
         assert len(results) == num_machines
 
+    @markers.aws.needs_fixing
+    def test_state_machine_status_filter(self, aws_client):
```

Comment:
The test suite is transitioning from utilising 'assertion tests' present in this file to exclusively relying on 'snapshot tests', which record and persist AWS's behaviour, for LocalStack to be directly compared to.
Since these chances aim at supporting an api action, it should be moved to `tests.aws.services.stepfunctions.v2.test_sfn_api.TestSnfApi`, where there also are examples of snapshot tests. Examples in these scenarios also use the expected test fixtures, such as `create_state_machine`, which cleanup AWS resources after execution.
In similar test suites, such as `tests.aws.services.stepfunctions.v2.scenarios.test_base_scenarios.TestBaseScenarios`, utility functions such as `tests.aws.services.stepfunctions.utils.create_and_record_execution` are used to reduce the complexity of launching and tracking execution events.

Reply:
migrated